openmrs/omod/src/main/webapp/profiles.jsp

```diff
@@ -2,104 +2,155 @@
-  .selected { font-weight: bold; }
-  .none { color: #999; }
-  .filename { font-family: monospace; font-size: 16px; }
-  .profile-select { white-space: nowrap; }
-  select { vertical-align: top; width: 50em; }
-  input { font-size: 16px; }
+    .section {
+        margin: 2em 0;
+    }
+    div.message {
+        padding: 1em;
+        background: #ffe;
+        border: 1px solid #aa5;
+    }
+    div.error.message {
+        background: #fee;
+        border: 1px solid #a55;
+    }
+    .output {
+        background: #ccc;
+        padding: 1em;
+    }
+    .selected {
+        font-weight: bold;
+    }
+    .none {
+        color: #999;
+    }
+    .filename {
+        font-family: monospace;
+        font-size: 16px;
+    }
+    .profile-select {
+        white-space: nowrap;
+    }
+    select {
+        vertical-align: top;
+        width: 50em;
+        margin-left: 8px;
+    }
+    input {
+        font-size: 16px;
+    }
+    #upload > span  {
+        position: relative;
+    }
+    #upload > span > input[type="file"] {
+        position: absolute;
+        opacity: 0;
+    }
+    #upload > span > input[type="submit"] {
+        position: absolute;
+        margin-left: 30px;
+    }
+    div.section.profile-current > span {
+        margin-left: 10px;
+    }
 </style>
 
 <h3>Project Buendia profiles</h3>
 
 <c:if test="${not authorized}">
-  <c:set var="disabledIfUnauthorized" value="disabled"/>
-  <div class="section error message">
-    You don't have the necessary privileges to manage profiles.
-    Please <a href="/openmrs/login.htm">log in</a> as a user authorized to
-    <b>Manage Concepts</b> and <b>Manage Forms</b>.
-  </div>
+    <c:set var="disabledIfUnauthorized" value="disabled"/>
+    <div class="section error message">
+        You don't have the necessary privileges to manage profiles.
+        Please <a href="/openmrs/login.htm">log in</a> as a user authorized to
+        <b>Manage Concepts</b> and <b>Manage Forms</b>.
+    </div>
 </c:if>
 
-<c:if test="${not empty param.failure}">
-  <div class="section error message">
-    The profile <span class="filename">${fn:escapeXml(param.filename)}</span>
-    could not be
-    <c:if test="${param.failure == 'add'}">added.</c:if>
-    <c:if test="${param.failure == 'apply'}">applied.</c:if>
-    <c:if test="${param.failure == 'delete'}">deleted.</c:if>
-    <pre class="output">${fn:escapeXml(param.output)}</pre>
-    Please correct the problems and try again.
-  </div>
-</c:if>
-
-<c:if test="${not empty param.success}">
-  <div class="section message">
-    <c:if test="${param.success == 'add'}">Added profile</c:if>
-    <c:if test="${param.success == 'apply'}">Applied profile</c:if>
-    <c:if test="${param.success == 'delete'}">Deleted profile</c:if>
-    <span class="filename">${fn:escapeXml(param.filename)}</span>.
-  </div>
+<c:if test="${not empty message}">
+    <div class="section <c:if test="${success == 'false'}">error</c:if> message">
```

Comment:
Great! This was before I changed params to models. I forgot to change this. Well done.